File: http-process/src/main/java/com/github/dragonhht/http/response/ResponseFacade.java
package com.github.dragonhht.http.response;

import javax.servlet.ServletOutputStream;
import javax.servlet.ServletResponse;
import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.Collection;
import java.util.Locale;

/**
 * 使用外观类，屏蔽掉Response中特有的方法，防止servlet类中的service方法调用.
 *
 * @author: huang
 * Date: 2018/4/20
 */
public class ResponseFacade implements HttpServletResponse {

    private Response response;

    public ResponseFacade(Response response) {
        this.response = response;
    }

    @Override
    public String getCharacterEncoding() {
        return this.response.getCharacterEncoding();
    }

    @Override
    public String getContentType() {
        return this.response.getContentType();
    }

    @Override
    public ServletOutputStream getOutputStream() throws IOException {
        return this.response.getOutputStream();
    }

    @Override
    public PrintWriter getWriter() throws IOException {
        return this.response.getWriter();
    }

    @Override
    public void setCharacterEncoding(String charset) {
        this.response.setCharacterEncoding(charset);
    }

    @Override
    public void setContentLength(int len) {
        this.response.setContentLength(len);
    }

    @Override
    public void setContentLengthLong(long len) {
        this.response.setContentLengthLong(len);
    }

    @Override
    public void setContentType(String type) {
        this.response.setContentType(type);
    }

    @Override
    public void setBufferSize(int size) {
        this.response.setBufferSize(size);
    }

    @Override
    public int getBufferSize() {
        return this.response.getBufferSize();
    }

    @Override
    public void flushBuffer() throws IOException {
        this.response.flushBuffer();
    }

    @Override
    public void resetBuffer() {
        this.response.resetBuffer();
    }

    @Override
    public boolean isCommitted() {
        return this.response.isCommitted();
    }

    @Override
    public void reset() {
        this.response.reset();
    }

    @Override
    public void setLocale(Locale loc) {
        this.response.setLocale(loc);
    }

    @Override
    public Locale getLocale() {
        return this.response.getLocale();
    }

    @Override
    public void addCookie(Cookie cookie) {
        this.response.addCookie(cookie);
    }

    @Override
    public boolean containsHeader(String s) {
        return this.response.containsHeader(s);
    }

    @Override
    public String encodeURL(String s) {
        return this.response.encodeURL(s);
    }

    @Override
    public String encodeRedirectURL(String s) {
        return this.response.encodeRedirectURL(s);
    }

    @Override
    public String encodeUrl(String s) {
        return this.response.encodeUrl(s);
    }

    @Override
    public String encodeRedirectUrl(String s) {
        return this.response.encodeRedirectUrl(s);
    }

    @Override
    public void sendError(int i, String s) throws IOException {
        this.response.sendError(i, s);
    }

    @Override
    public void sendError(int i) throws IOException {
        this.response.sendError(i);
    }

    @Override
    public void sendRedirect(String s) throws IOException {
        this.response.sendRedirect(s);
    }

    @Override
    public void setDateHeader(String s, long l) {
        this.response.setDateHeader(s, l);
    }

    @Override
    public void addDateHeader(String s, long l) {
        this.response.addDateHeader(s, l);
    }

    @Override
    public void setHeader(String s, String s1) {
        this.response.setHeader(s, s1);
    }

    @Override
    public void addHeader(String s, String s1) {
        this.response.addHeader(s, s1);
    }

    @Override
    public void setIntHeader(String s, int i) {
        this.response.setIntHeader(s, i);
    }

    @Override
    public void addIntHeader(String s, int i) {
        this.response.addIntHeader(s, i);
    }

    @Override
    public void setStatus(int i) {
        this.response.setStatus(i);
    }

    @Override
    public void setStatus(int i, String s) {
        this.response.setStatus(i, s);
    }

    @Override
    public int getStatus() {
        return this.response.getStatus();
    }

    @Override
    public String getHeader(String s) {
        return this.response.getHeader(s);
    }

    @Override
    public Collection<String> getHeaders(String s) {
        return this.response.getHeaders(s);
    }

    @Override
    public Collection<String> getHeaderNames() {
        return this.response.getHeaderNames();
    }
}

File: netty-server/src/main/java/com/github/dragonhht/http/servlet/ResponseFacade.kt
package com.github.dragonhht.http.servlet

import java.io.IOException
import java.io.PrintWriter
import java.util.*
import javax.servlet.ServletOutputStream
import javax.servlet.http.Cookie
import javax.servlet.http.HttpServletResponse

/**
 * .
 *
 * @author: huang
 * @Date: 2019-4-17
 */
class ResponseFacade(private val response: Response) : HttpServletResponse {

    override fun getCharacterEncoding(): String? {
        return this.response.characterEncoding
    }

    override fun getContentType(): String? {
        return this.response.contentType
    }

    @Throws(IOException::class)
    override fun getOutputStream(): ServletOutputStream? {
        return this.response.outputStream
    }

    @Throws(IOException::class)
    override fun getWriter(): PrintWriter? {
        return this.response.writer
    }

    override fun setCharacterEncoding(charset: String) {
        this.response.setCharacterEncoding(charset)
    }

    override fun setContentLength(len: Int) {
        this.response.setContentLength(len)
    }

    override fun setContentLengthLong(len: Long) {
        this.response.setContentLengthLong(len)
    }

    override fun setContentType(type: String) {
        this.response.setContentType(type)
    }

    override fun setBufferSize(size: Int) {
        this.response.bufferSize = size
    }

    override fun getBufferSize(): Int {
        return this.response.bufferSize
    }

    @Throws(IOException::class)
    override fun flushBuffer() {
        this.response.flushBuffer()
    }

    override fun resetBuffer() {
        this.response.resetBuffer()
    }

    override fun isCommitted(): Boolean {
        return this.response.isCommitted
    }

    override fun reset() {
        this.response.reset()
    }

    override fun setLocale(loc: Locale) {
        this.response.setLocale(loc)
    }

    override fun getLocale(): Locale? {
        return this.response.locale
    }

    override fun addCookie(cookie: Cookie) {
        this.response.addCookie(cookie)
    }

    override fun containsHeader(s: String): Boolean {
        return this.response.containsHeader(s)
    }

    override fun encodeURL(s: String): String? {
        return this.response.encodeURL(s)
    }

    override fun encodeRedirectURL(s: String): String? {
        return this.response.encodeRedirectURL(s)
    }

    override fun encodeUrl(s: String): String? {
        return this.response.encodeUrl(s)
    }

    override fun encodeRedirectUrl(s: String): String? {
        return this.response.encodeRedirectUrl(s)
    }

    @Throws(IOException::class)
    override fun sendError(i: Int, s: String) {
        this.response.sendError(i, s)
    }

    @Throws(IOException::class)
    override fun sendError(i: Int) {
        this.response.sendError(i)
    }

    @Throws(IOException::class)
    override fun sendRedirect(s: String) {
        this.response.sendRedirect(s)
    }

    override fun setDateHeader(s: String, l: Long) {
        this.response.setDateHeader(s, l)
    }

    override fun addDateHeader(s: String, l: Long) {
        this.response.addDateHeader(s, l)
    }

    override fun setHeader(s: String, s1: String) {
        this.response.setHeader(s, s1)
    }

    override fun addHeader(s: String, s1: String) {
        this.response.addHeader(s, s1)
    }

    override fun setIntHeader(s: String, i: Int) {
        this.response.setIntHeader(s, i)
    }

    override fun addIntHeader(s: String, i: Int) {
        this.response.addIntHeader(s, i)
    }

    override fun setStatus(i: Int) {
        this.response.status = i
    }

    override fun setStatus(i: Int, s: String) {
        this.response.setStatus(i, s)
    }

    override fun getStatus(): Int {
        return this.response.status
    }

    override fun getHeader(s: String): String? {
        return this.response.getHeader(s)
    }

    override fun getHeaders(s: String): Collection<String>? {
        return this.response.getHeaders(s)
    }

    override fun getHeaderNames(): Collection<String>? {
        return this.response.headerNames
    }
}